simSPI/linear_simulator/params_utils.py

```diff
@@ -0,0 +1,352 @@
+    def get_params(self):
+        """Get the parameters."""
+
+
+"""Module to generate parameters using the input star file."""
+
+
+class starfile_params(Iparams):
+    """Class to generate parameters using the input star file.
+
+    Parameters
+    ----------
+    config: class
+        Class containing parameters of the dataset generator.
+
+    """
+
+    def __init__(self, config):
+
+        self.counter = 0
+        self.invert_hand = False
+        self.df = starfile.read(config.input_starfile_path)
+        check_star_file(config.input_starfile_path)
+        self.config = config
+        print("Reading parameters from the input starfile.")
+
+    def get_params(self):
+        """Get the parameters from the starfile.
+
+        Returns
+        -------
+        rot_params: dict of type str to {tensor}
+            Dictionary of rotation parameters for a projection chunk
+        ctf_params: dict of type str to {tensor}
+            Dictionary of Contrast Transfer Function (CTF) parameters
+            for a projection chunk
+        shift_params: dict of type str to {tensor}
+            Dictionary of shift parameters for a projection chunk
+        """
+        self.particle = self.df["particles"].iloc[
+            self.counter : self.counter + self.config.chunks
+        ]
+        self.counter += self.config.chunks
+        return self.get_rotmat(), self.get_ctf_params(), self.get_shift_params()
+
+    def get_rotmat(self):
+        """Get the parameters for the rotation of the projections from starfile.
+
+        Returns
+        -------
+        rotmat_params: dict of type str to {tensor}
+            dictionary containing:
+            "rotmat": torch.Tensor
+                Tensor of size (chunks,3,3) that containing the rotation matrix for
+                each projection
+            "relion_AnglePsi": torch.Tensor
+                Tensor of size (chunks,) that contains the Psi angle (in degrees)
+                 for ZYZ rotation
+            "relion_AngleTilt": torch.Tensor
+                Tensor of size (chunks,) that contains the Tilt angle (in degrees)
+                 for ZYZ rotation
+            "relion_AngleRot": torch.Tensor
+                Tensor of size (chunks,) that contains the Rot angle (in degrees)
+                for ZYZ rotation
+        """
+        AnglePsi = np.array(self.particle["rlnAnglePsi"], ndmin=2).squeeze()
+        AngleTilt = np.array(self.particle["rlnAngleTilt"], ndmin=2).squeeze()
+        AngleRot = np.array(self.particle["rlnAngleRot"], ndmin=2).squeeze()
+        relion_euler_np = np.radians(
+            np.stack(
+                [
+                    -AnglePsi,  # convert Relion to our convention
+                    AngleTilt * (-1 if self.config.relion_invert_hand else 1),
+                    # convert Relion to our convention + invert hand
+                    AngleRot,
+                ]
+            )
+        )  # convert Relion to our convention
```

Comment:
I think this should be modularized outside, so that there's no upstream program specific conversions done in the code, but all in one place upstream.